src/tim/vx/tensor.cc

```diff
@@ -167,7 +183,79 @@ bool TensorImpl::CopyDataFromTensor(void* data) {
   return retn;
 }
 
-bool TensorImpl::Init() {
+bool TensorImpl::FlushCacheForHandle() {
+  if (!(spec_.attr_ & TensorAttribute::INPUT)) {
+    return false;
+  }
+
+  bool retn = true;
+  if (VSI_NN_TENSOR_ID_NA != id_) {
+    retn = false;
+    vsi_nn_tensor_t* tensor = vsi_nn_GetTensor(graph_->graph(), id_);
+    if (tensor && tensor->attr.is_created_from_handle) {
+      retn = (VSI_SUCCESS == vsi_nn_FlushHandle(tensor));
+      if (!retn) {
+        VSILOGE("FlushHandle fail");
+      }
+    }
+  }
+  return retn;
+}
+
+bool TensorImpl::RefillCacheFromHandle() {
+  if (!(spec_.attr_ & TensorAttribute::OUTPUT)) {
+    return false;
+  }
+
+  bool retn = true;
+  if (VSI_NN_TENSOR_ID_NA != id_) {
+    retn = false;
+    vsi_nn_tensor_t* tensor = vsi_nn_GetTensor(graph_->graph(), id_);
+    if (tensor && tensor->attr.is_created_from_handle) {
+      void* ptr = NULL;
+      retn = (VSI_SUCCESS == vsi_nn_GetTensorHandle(tensor, &ptr));
+      if (!retn) {
+        VSILOGE("GetTensorHandle fail");
+      }
+    }
+  }
+  return retn;
+}
+
+void* TensorImpl::map(bool refill_cache) {
+  if (!(spec_.attr_ & (TensorAttribute::INPUT | TensorAttribute::OUTPUT))) {
+    return nullptr;
+  }
+
+  void* cpu_ptr = nullptr;
+  if (VSI_NN_TENSOR_ID_NA != id_) {
+    vsi_nn_tensor_t* tensor = vsi_nn_GetTensor(graph_->graph(), id_);
+    if (tensor && tensor->attr.is_created_from_handle) {
+      if (data_ && !refill_cache) {
+        cpu_ptr = data_;
+      } else {
+        vsi_nn_GetTensorHandle(tensor, &cpu_ptr);
+        // TODO: what to do when fd_ != -1
```

Comment:
If the "managed" data is from fd, we can just use mmap() to retrieve the memory pointer.

Reply:
Um, because I don't know much about `fd_` or how to create a valid `fd`, I can not implement it correctly. Could you modify this PR directly? Or may we leave it as is? Then you official developers may replace it in following development.

Reply:
Okay, You can just keep it as TODO. We can follow-up later.